launcher/src/editor-configurations.ts

```diff
@@ -0,0 +1,110 @@
+/**********************************************************************
+ * Copyright (c) 2024 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ ***********************************************************************/
+
+import * as k8s from '@kubernetes/client-node';
+import * as fs from './fs-extra.js';
+
+const CONFIGMAP_NAME = 'editor-configurations';
+
+const enum EditorConfigs {
+  Settings = 'settings',
+  Extensions = 'extensions',
+}
+
+export class EditorConfigurations {
+  private coreV1API: k8s.CoreV1Api;
+
+  constructor(private readonly workspaceFile: string) {}
+
+  async configure(): Promise<void> {
+    console.log('# Checking if editor configurations are provided...');
+
+    const configmap = await this.getConfigmap();
+    if (!configmap || !configmap.data) {
+      console.log('  > Editor configurations are not provided');
+      return;
+    }
+
+    console.log('  > Editor configurations are provided, trying to apply them...');
+
+    const workspaceFileContent = await fs.readFile(this.workspaceFile);
+    const existingData = parseJsonFrom(workspaceFileContent);
+    if (!existingData) {
+      console.log(`  > Can not apply editor configurations: failed to parse data from the ${this.workspaceFile}`);
+      return;
+    }
+
+    this.configureSection(EditorConfigs.Settings, configmap, existingData);
+    this.configureSection(EditorConfigs.Extensions, configmap, existingData);
+
+    fs.writeFile(this.workspaceFile, JSON.stringify(existingData, null, '\t'));
```

Comment:
TODO: add check if changes where applied and only for this case `writing file` operation should be applied